### PR TITLE
fix(can): Wait for a free slot in the TX message fifo before trying sending.

### PR DESCRIPTION
**Pull request by amitlissack**

### CODE REVIEW

**Review by DerekMaggio — APPROVED** [review]
Lgtm

File: can/firmware/hal_can.c [hunk 1]
```diff
@@ -100,6 +100,11 @@ void can_send_message(HAL_CAN_HANDLE handle, uint32_t arbitration_id, uint8_t* b
         .TxEventFifoControl = FDCAN_NO_TX_EVENTS,
         .MessageMarker=0
     };
+
+    // Wait for there to be room.
+    while (HAL_FDCAN_GetTxFifoFreeLevel(handle) == 0)
```
Comment (sfoster1) [inline can/firmware/hal_can.c:105]: Can we [vTaskDelay(1)](https://www.freertos.org/a00127.html) here, little scared of a busyloop like this

**Review by sfoster1 — CHANGES_REQUESTED** [review]
Can we delay instead of straight up looping? Otherwise looks great, will test on hardware today

**Review by sfoster1 — APPROVED** [review]
Looks good to me! Works great on hardware

**Pull request merged**